File: Java/OLD/MultipleInheritance.java
interface A {
	public final int a=10;
	public void show();
}

interface B {
	public final int b=20;
	public void show();
}


class C implements A, B {
	
	private int c;
	public C() {
		c=30;
	}
	
		public void show() {					//override
			System.out.println("a=" + a);
			System.out.println("b=" + b);
			System.out.println("c=" + c);
		}
}

class MultipleInheritanceTest
{
	public static void main(String[] args)
	{
		C obc = new C();
	
		obc.show();
	}
}

File: CPP/MultipleInheritance.cpp
#include<iostream>

using namespace std;

class A
{
		int a;
	public:
		A()
		{
			a=10;
		}
		void show();
};

void A :: show()
{
	cout<<"a="<<a<<endl;
}

class B
{
		int b;
	public:
		B()
		{
			b=20;
		}
		void show();
};

void B :: show()
{
	cout<<"b="<<b<<endl;
}

class C : public A, public B
{
		int c;
	public:
		C()
		{
			c=30;
		}
		void showC();
};

void C :: showC()					//Overriding
{
	cout<<"c="<<c<<endl;
}


int main()
{
	C obc;
	
	obc.A::show();
	obc.B::show();
	obc.showC();
	
	return 0;
}

